cmd/flux/check.go

```diff
@@ -77,187 +99,298 @@ func init() {
 	rootCmd.AddCommand(checkCmd)
 }
 
-func runCheckCmd(cmd *cobra.Command, args []string) error {
-	logger.Actionf("checking prerequisites")
-	checkFailed := false
-
-	fluxCheck()
-
+func runCheckCmd(_ *cobra.Command, _ []string) error {
 	ctx, cancel := context.WithTimeout(context.Background(), rootArgs.timeout)
 	defer cancel()
 
 	cfg, err := utils.KubeConfig(kubeconfigArgs, kubeclientOptions)
 	if err != nil {
-		return fmt.Errorf("Kubernetes client initialization failed: %s", err.Error())
+		return fmt.Errorf("kubernetes client initialization failed: %w", err)
 	}
 
 	kubeClient, err := client.New(cfg, client.Options{Scheme: utils.NewScheme()})
 	if err != nil {
-		return err
+		return fmt.Errorf("error creating kubernetes client: %w", err)
 	}
 
-	if !kubernetesCheck(cfg, kubernetesConstraints) {
-		checkFailed = true
+	if !runPreChecks(ctx, cfg) {
+		return errors.New("pre-installation checks failed")
```

Comment:
If `--pre` was not passed, we should continue with the other checks and not exit here.